pull_request_package/lib/github_status_reporter.rb

```diff
@@ -3,22 +3,57 @@
 
 class GitHubStatusReporter
   include ActiveModel::Model
-  attr_accessor :client, :logger, :pull_request
+  attr_accessor :client, :logger, :package
   
   def report
-    client.create_status('openSUSE/open-build-service', 'f73658fa927b90a0a3d28b79866215210b856390', 'pending', options) if [:success, :failed].include?(state)
+    summary = self.summary
+    state = :success
+    if summary[:failure] > 0
+      state = :failure
+    elsif summary[:pending] > 0 or summary[:success] == 0
+      state = :pending
+    end
+    count = summary[:success] + summary[:failure] + summary[:pending]
+    case state
+    when :failure
+       descr = "#{summary[:failure]}/#{count} failed"
+    when :success
+       descr = "#{count} succeeded"
+    else
+       descr = "#{summary[:pending]}/#{count} building"
+    end
```

Comment:
Maybe you can move line 10 - 15 to a dedicated function state? And line 16 - 24 to a dedicated function count.